File: in_java/Pattern_9.java
package in_java;

public class Pattern_9 extends Pattern {
    public Pattern_9(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        char symbol = this.symbol.charAt(0);
        for (int i = 0; i < this.row; i++) {
            symbol = this.symbol.charAt(0);
            for (int j = 0; j < this.column; j++) {
                if (j <= row - 1 - i || j >= row - 1 + i) {
                    System.out.print(symbol);
                } else {
                    System.out.print(" ");
                }
                if (j < row - 1) {
                    symbol++;
                } else {
                    symbol--;
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_9.rs
pub fn pattern_9() {
    let row = 5;
    let column = 9;
    let mut symbol: u8;

    for i in 0..row {
        symbol = 65;
        for j in 0..column {
            if j <= row - 1 - i || j >= row - 1 + i {
                print!("{}", symbol as char);
            } else {
                print!(" ");
            }
            if j < row - 1 {
                symbol += 1;
            } else {
                symbol -= 1;
            }
        }
        println!();
    }
}
